vercel/turborepo
# add method for determining base commit on GitHub actions

### Description

If you're running `--affected` and you're running in the context of a GitHub action there are some [additional env vars that are available](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables) that give more context (pull request action, push action, each have their own thing).  In a push even it's a previous commit, and in the pull request event it's the base of the pull request.  We want to use those to be the base of --affected instead of just defaulting to main or whatever we have.  You can still override it as normal.

This is a problem right now for running in GitHub actions right now because it clones to a limited depth (shallow clone).  We might therefore need to check that the commit in question actually exists.

### Testing Instructions
 
see tests (there are quite a few) - ultimately the only way to "really" test this is to see it run in GitHub runners.
